src/room/usePTT.ts

```diff
@@ -39,6 +63,24 @@ export const usePTT = (
   userMediaFeeds: CallFeed[],
   playClip: PlayClipFunction
 ): PTTState => {
+  // Used to serialise all the mute calls so they don't race. It has
+  // its own state as its always set separately from anything else.
+  const [mutePromise, setMutePromise] = useState(Promise.resolve());
+
+  // Wrapper to serialise all the mute operations on the promise
+  const setMicMuteWrapper = useCallback(
+    (muted) => {
+      setMutePromise(
+        mutePromise.then(() => {
+          groupCall.setMicrophoneMuted(muted).catch((e) => {
```

Comment:
Are you supposed to be returning this promise here so that it properly awaits the `setMicrophoneMuted` promise?

Reply:
Yes! Good catch.